shell/components/dialog/ConfirmNameToRemoveDialog.vue

```diff
@@ -0,0 +1,108 @@
+<script>
+import { mapGetters } from 'vuex';
+import { exceptionToErrorsArray } from '@shell/utils/error';
+import { alternateLabel } from '@shell/utils/platform';
+import AsyncButton from '@shell/components/AsyncButton';
+import { Banner } from '@components/Banner';
+import { Card } from '@components/Card';
+import CopyToClipboardText from '@shell/components/CopyToClipboardText';
+
+/**
+ * @name ConfirmNameToRemoveDialog
+ * @description Dialog component to confirm name before removing the resource.
+ */
+export default {
+  name: 'ConfirmNameToRemoveDialog',
+
+  components: {
+    AsyncButton,
+    Banner,
+    Card,
+    CopyToClipboardText
+  },
+
+  props:      {
+    resources: {
+      type:     Array,
+      required: true
+    }
+  },
+
+  data() {
+    return { errors: [], confirmName: '' };
+  },
+
+  computed: {
+    ...mapGetters({ t: 'i18n/t' }),
+
+    nameToMatch() {
+      return this.resources[0].nameDisplay;
+    },
+
+    deleteDisabled() {
+      const confirmFailed = this.confirmName !== this.nameToMatch;
+
+      return this.preventDelete || confirmFailed;
+    },
+
+    protip() {
+      return this.t('promptRemove.protip', { alternateLabel });
+    },
+  },
+
+  methods: {
+    close() {
+      this.confirmName = '';
+      this.errors = [];
+      this.$emit('close');
+    },
+
+    async remove(buttonDone) {
+      try {
+        await this.resources[0].remove();
```

Comment:
Could we still call it `delete`, since everything about it is for deleting, while we just use a method to remove, but the whole scope is to delete something?

Reply:
If this is a method I just created I would be happy to do it. This is the *standard* method in the code base to *delete*. This model extends `SteveModel` and overrides its `remove` method. So I am apprehensive to deviate from the convention in the code base in this instance as part of this PR.